src/main/java/org/zalando/nakadi/service/ConsumerLimitingService.java

```diff
@@ -0,0 +1,175 @@
+package org.zalando.nakadi.service;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.echocat.jomon.runtime.concurrent.RetryForSpecifiedCountStrategy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.zalando.nakadi.exceptions.ConnectionSlotOccupiedException;
+import org.zalando.nakadi.exceptions.NakadiRuntimeException;
+import org.zalando.nakadi.exceptions.NoConnectionSlotsException;
+import org.zalando.nakadi.exceptions.ServiceUnavailableException;
+import org.zalando.nakadi.repository.zookeeper.ZooKeeperHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+import static java.text.MessageFormat.format;
+import static org.echocat.jomon.runtime.concurrent.Retryer.executeWithRetry;
```

Comment:
Hell no! :)
We had a balance of -1 in our radar which doesn't allow neither adopt nor forbid that.